README.md
<div align="center">

![lanterngc-icon](img/favicon-192x192.png) \
[![Download lanterngc](https://a.fsdn.com/con/app/sf-download-button)](https://sourceforge.net/projects/lanterngc/files/latest/download) \
![Download lanterngc](https://img.shields.io/sourceforge/dt/lanterngc.svg) 

# GC-Lantern

    A simple and practical lantern for your desktop/tablet computer

## Graphical Interface

![demo-gui](img/lanterngc.png)

---

&copy; 2022 Nurul-GC \
&trade; ArtesGC

</div>


CHANGELOG.md
# LGC - CHANGELOG

- [v01-052022](https://github.com/Nurul-GC/lanterngc/releases/tag/v0.1)
  - Idea's Creation and organization;
  - Customized the layout and the principal functions;
- [v02-062022](https://github.com/Nurul-GC/lanterngc/releases/tag/v0.2)
  - Changed the icons;
  - Optimized the theme;
  - Fixed some minor bugs;


gcl.py
import webbrowser
from sys import argv, exit

from PyQt6.QtWidgets import *
from PyQt6.QtGui import *
from PyQt6.QtCore import *


class GCL:
    def __init__(self):
        self.gc = QApplication(argv)
        QFontDatabase.addApplicationFont('fonts/Kranky.ttf')

        self.ferramentas = QWidget()
        self.ferramentas.setWindowTitle('GC-lantern')
        self.ferramentas.setWindowState(Qt.WindowState.WindowMaximized)
        self.ferramentas.setWindowFlags(Qt.WindowType.WindowMinimizeButtonHint)
        self.ferramentas.setStyleSheet(temaescuro)

        self.layout = QVBoxLayout()

        menu = QMenuBar()
        detalhes = menu.addMenu('Details')
        instr = detalhes.addAction('Instructions')
        instr.triggered.connect(self._instr)
        detalhes.addSeparator()
        sair = detalhes.addAction('Exit')
        sair.triggered.connect(self._sair)
        sobre = menu.addAction('About')
        sobre.triggered.connect(self._sobre)
        self.layout.setMenuBar(menu)
        self.ferramentas.setLayout(self.layout)

        self.principal()

        link = lambda: webbrowser.open_new('https://artesgc.home.blog')
        copyrightlabel = QLabel('<a style="text-decoration: none;" href="#">&trade; ArtesGC, Inc</a>')
        copyrightlabel.setAlignment(Qt.AlignmentFlag.AlignCenter)
        copyrightlabel.linkActivated.connect(link)
        copyrightlabel.setToolTip("Open ArtesGC's oficial website!")
        barramenu = QToolBar("Copyright")
        barramenu.setLayoutDirection(Qt.LayoutDirection.RightToLeft)
        barramenu.addWidget(copyrightlabel)
        self.layout.addWidget(barramenu)

    def principal(self):
        def ligar():
            if lantern_btn.isChecked():
                lantern_btn.setText('OFF')
                self.ferramentas.setStyleSheet(temaclaro)
            else:
                lantern_btn.setText('ON')
                self.ferramentas.setStyleSheet(temaescuro)

        frame = QFrame()
        layout = QVBoxLayout()

        lantern_btn = QPushButton('ON')
        lantern_btn.setFixedHeight(550)
        lantern_btn.setAutoExclusive(True)
        lantern_btn.setCheckable(True)
        lantern_btn.clicked.connect(ligar)
        layout.addWidget(lantern_btn)

        frame.setLayout(layout)
        self.layout.addWidget(frame)

    def _sobre(self):
        QMessageBox.information(self.ferramentas, 'About', '''<h2>Information about the Program</h2><hr>
        Name: <b>GC-lantern (GCL)</b><br>
        Version: <b>0.1-042022</b><br>
        Programmer & Designer: <b>Nurul-GC</b><br>
        Company: <b>ArtesGC, Inc.</b>''')

    def _instr(self):
        QMessageBox.information(self.ferramentas, 'Instructions', '''<h2>Quick Presentation</h2><hr>
        <p>Hello Dear User!<br>
        I present to you a very practical and useful tool,
        a program that works as a lantern on your desktop/tablet PC..</p>
        <p>It's easy to be used, you just need to push the button and it will change his theme,
        helping you to see what's around you!</p>
        <p>Thanks for your support!<br>
        <b>&trade;ArtesGC, Inc.</b></p>''')

    def _sair(self):
        pergunta = QMessageBox.question(self.ferramentas, 'Confirm Exit', 'Do you really want to close the program?')
        if pergunta == pergunta.Yes:
            exit(0)


if __name__ == '__main__':
    temaescuro = open('theme/dark.qss').read().strip()
    temaclaro = open('theme/light.qss').read().strip()
    app = GCL()
    app.ferramentas.show()
    app.gc.exec()


lgc/__init__.py
from random import randint
from sys import argv
from time import sleep

from PyQt6.QtCore import *
from PyQt6.QtGui import *
from PyQt6.QtWidgets import *

from lgcmain import LGC


class LGCINIT:
    def __init__(self):
        self.gc = QApplication(argv)

        # application font
        QFontDatabase.addApplicationFont("./lgc-fonts/Kranky.ttf")

        img = QPixmap("./lgc-icons/init.png")
        self.align = int(Qt.AlignmentFlag.AlignBottom | Qt.AlignmentFlag.AlignHCenter | Qt.AlignmentFlag.AlignAbsolute)
        self.color = Qt.GlobalColor.yellow
        self.janela = QSplashScreen(img)
        self.janela.setStyleSheet('font-family: "Kranky";'
                                  'font-size: 11pt;')
        self.janela.show()
        self.iniciar()

    def iniciar(self):
        load = 0
        while load < 100:
            self.janela.showMessage(f"Loading Package: {load}%", self.align, self.color)
            sleep(0.3)
            load += randint(5, 10)
        self.janela.close()
        app = LGC()
        app.ferramentas.show()


if __name__ == '__main__':
    lgcApp = LGCINIT()
    lgcApp.gc.exec()


lgc/lgcmain.py
from sys import argv, exit
import webbrowser

from PyQt6.QtCore import *
from PyQt6.QtGui import *
from PyQt6.QtWidgets import *

temaescuro = open('lgc-theme/dark.qss').read().strip()
temaclaro = open('lgc-theme/light.qss').read().strip()


class LGC:
    def __init__(self):
        # main-window-widget
        self.ferramentas = QWidget()
        self.ferramentas.setWindowTitle('LanternGC')
        self.ferramentas.setWindowState(Qt.WindowState.WindowMaximized)
        self.ferramentas.setWindowFlags(Qt.WindowType.WindowMinimizeButtonHint)
        self.ferramentas.setStyleSheet(temaescuro)

        # main-layout
        self.layout = QFormLayout()
        self.layout.setSpacing(10)

        # menu-bar
        menu = QMenuBar()
        detalhes = menu.addMenu('Details')
        instr = detalhes.addAction('Instructions')
        instr.triggered.connect(self._instr)
        detalhes.addSeparator()
        sair = detalhes.addAction('Exit')
        sair.triggered.connect(self._sair)
        sobre = menu.addAction('About')
        sobre.triggered.connect(self._sobre)
        self.layout.setMenuBar(menu)
        self.ferramentas.setLayout(self.layout)

        # main-button-function
        def ligar():
            if self.lantern_btn.isChecked():
                self.ferramentas.setStyleSheet(temaclaro)
            else:
                self.ferramentas.setStyleSheet(temaescuro)

        self.lantern_btn = QPushButton()
        self.lantern_btn.setFixedHeight(550)
        self.lantern_btn.setCheckable(True)
        self.lantern_btn.setStyleSheet('''
        background-image: url("./lgc-icons/lantern.png");
        background-repeat: no-repeat;
        background-size: auto;
        background-attachment: fixed;
        background-position: center;
        ''')
        self.lantern_btn.clicked.connect(ligar)
        self.layout.addRow(self.lantern_btn)

        # copyright-label
        link = lambda: webbrowser.open_new('https://artesgc.home.blog')
        copyrightlabel = QLabel('<a style="text-decoration: none; background-color: none;" href="#">&trade;ArtesGC, Inc</a>')
        copyrightlabel.setAlignment(Qt.AlignmentFlag.AlignCenter)
        copyrightlabel.linkActivated.connect(link)
        copyrightlabel.setToolTip("Open ArtesGC's oficial website!")

        barramenu = QToolBar("Copyright")
        barramenu.setLayoutDirection(Qt.LayoutDirection.RightToLeft)
        barramenu.setToolButtonStyle(Qt.ToolButtonStyle.ToolButtonFollowStyle)
        barramenu.addWidget(copyrightlabel)
        self.layout.addRow(barramenu)

    # menu-bar-functions
    def _sobre(self):
        QMessageBox.information(self.ferramentas, 'About', '''<h2>Information about the Program</h2><hr>
        Name: <b>GC-lantern (LGC)</b><br>
        Version: <b>0.2-062022</b><br>
        Programmer & Designer: <b>Nurul-GC</b><br>
        Company: <b>ArtesGC, Inc.</b>''')

    def _instr(self):
        QMessageBox.information(self.ferramentas, 'Instructions', '''<h2>Quick Presentation</h2><hr>
        <p>Hello Dear User!<br>
        I present to you a very practical and useful tool,
        a program that works as a lantern on your desktop/tablet PC..</p>
        <p>It's easy to be used, you just need to push the button and it will change his lgc-theme,
        helping you to see what's around you!</p>
        <p>Thanks for your support!<br>
        <b>&trade;ArtesGC, Inc.</b></p>''')

    def _sair(self):
        if self.lantern_btn.isChecked():
            pergunta = QMessageBox.question(self.ferramentas, 'Confirm Exit', '<b>Do you really want to close the program?</b>')
            if pergunta == pergunta.Yes:
                return exit(0)
        else:
            return exit(0)
